crates/fluss/src/io/file_io.rs

```diff
@@ -97,12 +97,10 @@ impl FileIOBuilder {
     }
 }
 
-#[async_trait::async_trait]
 pub trait FileRead: Send + Unpin + 'static {
-    async fn read(&self, range: Range<u64>) -> Result<Bytes>;
+    fn read(&self, range: Range<u64>) -> impl Future<Output = Result<Bytes>> + Send;
 }
```

Comment:
See https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits/#async-fn-in-public-traits